traffic_portal/app/src/common/modules/form/deliveryServiceStaticDnsEntry/form.deliveryServiceStaticDnsEntry.tpl.html

```diff
@@ -50,9 +50,9 @@
             <div class="form-group" ng-class="{'has-error': hasError(dsStaticDnsEntryForm.ttl), 'has-feedback': hasError(dsStaticDnsEntryForm.ttl)}">
                 <label for="ttl" class="control-label col-md-2 col-sm-2 col-xs-12">TTL *</label>
                 <div class="col-md-10 col-sm-10 col-xs-12">
-                    <input id="ttl" name="ttl" type="number" class="form-control" ng-model="staticDnsEntry.ttl" min="0" step="1" required>
+                    <input id="ttl" name="ttl" type="number" class="form-control" ng-model="staticDnsEntry.ttl" ng-pattern="/^\d+$/" required>
```

Comment:
Does `min="0" step="1"` not work?

Reply:
It works but ng-pattern seems to be a better approach.

Reply:
Why? `min` and `step` are native `input[type=number]` controls that work within the browser. `ng-pattern` not only adds an event listener, but uses userland scripting to do regular expression matching. It's slower and less idiomatic than the native controls.

Reply:
@ocket8888 - how about just `pattern` instead of `ng-pattern` since the pattern is static? `min` does not prevent the user from entering -1, for example. basically, i just want to make sure an alert shows up if the user enters a non-whole number.

Reply:
> _"min does not prevent the user from entering -1, for example._"

What do you mean? Like the user can literally type in "-1"? Because you can still do that with `ng-pattern`. Also, without a min, I can use the browser's native controls to enter an invalid value. I can hit the down arrow and get -1. Using `min` and `step` if I put in "-1" I still see the field marked as invalid.

![image](https://user-images.githubusercontent.com/6013378/90281394-ba5e7780-de29-11ea-9201-0a59b164bf00.png)